src/snowflake/connector/connection.py

```diff
@@ -267,8 +267,8 @@ def __init__(self, **kwargs):
         self._errorhandler = Error.default_errorhandler
         self._lock_converter = Lock()
         self.messages = []
-        self._async_sfqids = set()
-        self._done_async_sfqids = set()
+        self._async_sfqids = {}
```

Comment:
`dict` starts to preserve insertion order since 3.7, so using `reverse(dict.keys())` allows us to start checking the last submitted queries first.

Reply:
Missing type-hints on these definitions

Reply:
I think [deque](https://docs.python.org/3/library/collections.html#deque-objects)s would be better here, there's no up-side to storing the keys as the values too, but you double the memory we use to keep the order things are inserted in.

Reply:
You're right! :)